packages/material-ui/src/Container/Container.js

```diff
@@ -18,58 +54,40 @@ export const styles = (theme) => ({
       paddingLeft: theme.spacing(3),
       paddingRight: theme.spacing(3),
     },
-  },
-  /* Styles applied to the root element if `disableGutters={true}`. */
-  disableGutters: {
-    paddingLeft: 0,
-    paddingRight: 0,
-  },
-  /* Styles applied to the root element if `fixed={true}`. */
-  fixed: Object.keys(theme.breakpoints.values).reduce((acc, breakpoint) => {
-    const value = theme.breakpoints.values[breakpoint];
+    ...(styleProps.disableGutters && {
+      paddingLeft: 0,
+      paddingRight: 0,
+    }),
+  }),
+  ({ theme, styleProps }) =>
```

Comment:
Is there a reason why we don't have just one function callback as we do in the other components?

Reply:
Yes, to keep each breakpoint isolated, otherwise the last one erase the previous ones.